pkg/plugins/optional/helm/v1alpha/scaffolds/init.go

```diff
@@ -0,0 +1,210 @@
+
+type initScaffolder struct {
+	config config.Config
+
+	// fs is the filesystem that will be used by the scaffolder
+	fs machinery.Filesystem
+
+	// force indicates whether to scaffold controller files even if it exists or not
+	force bool
+}
+
+// NewInitHelmScaffolder returns a new Scaffolder for HelmPlugin
+// nolint: lll
+func NewInitHelmScaffolder(config config.Config, force bool) plugins.Scaffolder {
+	return &initScaffolder{
+		config: config,
+		force:  force,
+	}
+}
+
+// InjectFS implements cmdutil.Scaffolder
+func (s *initScaffolder) InjectFS(fs machinery.Filesystem) {
+	s.fs = fs
+}
+
+// Scaffold scaffolds the Helm chart with the necessary files.
+func (s *initScaffolder) Scaffold() error {
+	log.Println("Generating Helm Chart to distribute project")
+
+	// Initialize the map to store the extracted images from deploy-image plugin
+	deployImages := make(map[string]string)
+
+	// Detect and extract the deploy-image plugin configuration
+	pluginConfig := struct {
+		Resources []struct {
+			Kind    string            `json:"kind"`
+			Options map[string]string `json:"options"`
+		} `json:"resources"`
+	}{}
+
+	err := s.config.DecodePluginConfig("deploy-image.go.kubebuilder.io/v1-alpha", &pluginConfig)
+	if err != nil {
+		return fmt.Errorf("error decoding deploy-image plugin config: %w", err)
+	}
+
+	// Extract images for the resources
+	for _, res := range pluginConfig.Resources {
+		image, ok := res.Options["image"]
+		if ok {
+			deployImages[res.Kind] = image
+		}
+	}
+
+	// Initialize the machinery.Scaffold that will write the files to disk
+	scaffold := machinery.NewScaffold(s.fs,
+		machinery.WithConfig(s.config),
+	)
+
+	// Pass the extracted images to the HelmValues template
+	err = scaffold.Execute(
+		&templates.HelmChart{Force: s.force},
+		&templates.HelmValues{
+			HasWebhooks:  hasWebhooks(s.config),
+			DeployImages: deployImages,
+			Force:        s.force,
+		},
+		&templates.HelmIgnore{Force: s.force},
+		&chart_templates.HelmHelpers{Force: s.force},
+		&chart_templates.ManagerDeployment{Force: s.force},
+	)
+	if err != nil {
+		return err
+	}
+
+	// Copy all relevant files from config/ to chart/templates/
+	err = copyConfigFiles()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Helper function to copy files from config/ to dist/chart/templates/
+func copyConfigFiles() error {
+	// Define the source directories and destination directories under dist/chart/templates
+	configDirs := []struct {
+		SrcDir  string
+		DestDir string
+		SubDir  string
+	}{
+		{"config/rbac", "dist/chart/templates/rbac", "rbac"},
+		{"config/webhook", "dist/chart/templates/webhook", "webhook"},
+		{"config/crd/bases", "dist/chart/templates/crds", "crd"},
+		{"config/prometheus", "dist/chart/templates/prometheus", "prometheus"},
+		{"config/certmanager", "dist/chart/templates/certmanager", "certmanager"},
+		{"config/network-policy", "dist/chart/templates/network-policy", "networkPolicy"},
+	}
```

Comment:
It is just to getting started to implement it.
We probably need move out those to templates
Also, some changes might need to be required to be done in the content to use the labels from the helpers and use the namespace from values.